private-templates-service/client/src/store/templateOwner/reducers.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  OwnerState,
+  OwnerType,
+  GET_OWNER_NAME,
+  GET_OWNER_NAME_SUCCESS,
+  GET_OWNER_NAME_FAILURE,
+  GET_OWNER_PROFILE_PICTURE,
+  GET_OWNER_PROFILE_PICTURE_SUCCESS,
+  GET_OWNER_PROFILE_PICTURE_FAILURE,
+  GetOwnerNameAction,
+  GetOwnerProfilePictureAction,
+} from './types';
+
+const initialState: OwnerState = {
+  owner: undefined,
+  isFetching: false,
+}
+
+const initialOwnerState: OwnerType = {
+  displayName: "",
+}
+
+function createOwner(owner = initialOwnerState, action: GetOwnerNameAction | GetOwnerProfilePictureAction): OwnerType | undefined {
+  switch (action.type) {
+    case GET_OWNER_NAME_SUCCESS:
+      return {
+        ...owner,
+        displayName: action.ownerName
+      }
+    case GET_OWNER_PROFILE_PICTURE_SUCCESS:
+      return {
+        ...owner,
+        imageURL: action.ownerImageURL,
```

Comment:
Question about this.

It seems like each owner profile picture component is updating this single value. If there are multiple templates from different people does it show each person's different picture or just the last one fetched from the server? I would have expected the URLs to be stored in some sort of array or dict.

Reply:
If you refer to line 30 in RecentlyViewedTable.tsx, you will see the LastEditedUserInfo component (formerly known as OwnerInfo) is constructed for each Template in the RecentlyViewed array. This is all based on the oID in each Template in the RecentlyViewed. The person's profile picture should update accordingly.

Reply:
Right. I believe each user's image is fetched but the URL is stored at this location in the redux store. Each component looks at this element in the store to show the image. I still think the images may not be displayed correctly, but I think the best way to verify is to try it out. 

I tried to run it locally with two different accounts but I'm not able to see published templates from other users, preventing me from verifying this.

Reply:
Fixed it! 
![image](https://user-images.githubusercontent.com/40618774/77363473-050f1400-6d10-11ea-8f1a-b5d2a1c911bf.png)